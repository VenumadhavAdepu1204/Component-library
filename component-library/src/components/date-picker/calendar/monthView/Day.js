import React from 'react'

import { Days } from '../../styles'

const Day = props => {
  const { fullDate, onClick, selectedDate, onMouseEnter, onMouseLeave, hovering, disablePreviousDates, disableAfterDates } = props
  const date = fullDate && fullDate.getDate()
  const currentDate = new Date()
  if (disablePreviousDates) {
    if (fullDate < disablePreviousDates) {
      return <Days disabled data-test-id='day-disable-before'>
        {date}
      </Days>
    }
  }
  if (disableAfterDates) {
    if (fullDate > disableAfterDates) {
      return <Days disabled data-test-id='day-disable-after'>
        {date}
      </Days>
    }
  }

  if (hovering) {
    return <Days hovering
      data-test-id='day-hovering'
      onClick={() => onClick(fullDate)}
      onMouseEnter={() => onMouseEnter(date)}
      onMouseLeave={() => onMouseLeave(date)}>
      {date}
    </Days>
  } else if (fullDate.toDateString() === (selectedDate ? selectedDate.toDateString() : '')) {
    return <Days selected
      data-test-id='day-selected'
      onClick={() => onClick(fullDate)}
      onMouseEnter={() => onMouseEnter(date)}
      onMouseLeave={() => onMouseLeave(date)}>
      {date}
    </Days>
  } else if (fullDate.toDateString() === currentDate.toDateString()) {
    return <Days current
      data-test-id='day-current'
      onClick={() => onClick(fullDate)}
      onMouseEnter={() => onMouseEnter(date)}
      onMouseLeave={() => onMouseLeave(date)}>
      {date}
    </Days>
  } else {
    return (
      <Days
        data-test-id='day-remaining'
        onClick={() => onClick(fullDate)}
        onMouseEnter={() => onMouseEnter(date)}
        onMouseLeave={() => onMouseLeave(date)}>
        {date}
      </Days>
    )
  }
}

export default Day
