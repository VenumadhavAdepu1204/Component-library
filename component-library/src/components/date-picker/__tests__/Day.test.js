import React from 'react'
import { render, mount, configure } from 'enzyme'
import Day from '../calendar/monthView/Day'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

const mountApp = mount(<Day
  key={1}
  fullDate={new Date(2018, 10, 5)}
  onClick={jest.fn()}
  selectedDate={new Date(2018, 10, 7)}
  selected={false}
  hovering={false}
  onMouseEnter={jest.fn()}
  onMouseLeave={jest.fn()}
  focusTextInput={jest.fn()}
/>)

describe('Day ', () => {
  const fullDate = new Date(2018, 10, 5)
  const app = render(<Day
    key={1}
    fullDate={fullDate}
    onClick={jest.fn()}
    selectedDate={new Date(2018, 10, 7)}
    selected={false}
    hovering={false}
    onMouseEnter={jest.fn()}
    onMouseLeave={jest.fn()}
    disablePreviousDates={new Date(2018, 9, 7)}
    disableAfterDates={new Date(2018, 11, 7)}
    focusTextInput={jest.fn()}
  />)

  test('renders properly', () => {
    expect(app).toMatchSnapshot()
  })

  test('when a day is selected ', () => {
    mountApp.setProps({ hovering: false, selected: true, selectedDate: new Date(2018, 10, 5) })
    mountApp.find('[data-test-id="day-selected"]').at(0).simulate('mouseenter')
    mountApp.find('[data-test-id="day-selected"]').at(0).simulate('mouseleave')
    mountApp.find('[data-test-id="day-selected"]').at(0).simulate('click')
  })

  test('when a day is hovered ', () => {
    mountApp.setProps({ selected: false, hovering: true })
    mountApp.find('[data-test-id="day-hovering"]').at(0).simulate('mouseenter')
    mountApp.find('[data-test-id="day-hovering"]').at(0).simulate('click')
  })

  test('disable before and after', () => {
    mountApp.setProps({ fullDate: new Date(new Date(2018, 9, 6)) })
    mountApp.setProps({ fullDate: new Date(new Date(2018, 11, 8)) })
  })
})

describe('using mount', () => {
  test('for remaining days', () => {
    const wrapper = mount(<Day
      key={1}
      fullDate={new Date(2019, 4, 10)}
      onClick={jest.fn()}
      selectedDate={null}
      selected={false}
      hovering={false}
      onMouseEnter={jest.fn()}
      onMouseLeave={jest.fn()}
      focusTextInput={jest.fn()}
    />)
    wrapper.find('[data-test-id="day-remaining"]').at(0).simulate('mouseenter')
    wrapper.find('[data-test-id="day-remaining"]').at(0).simulate('mouseleave')
    wrapper.find('[data-test-id="day-remaining"]').at(0).simulate('click')
  })

  test('on current date', () => {
    const wrapper = mount(<Day
      key={1}
      fullDate={new Date()}
      onClick={jest.fn()}
      selectedDate={null}
      selected={false}
      hovering={false}
      onMouseEnter={jest.fn()}
      onMouseLeave={jest.fn()}
      focusTextInput={jest.fn()}
    />)
    wrapper.find('[data-test-id="day-current"]').at(0).simulate('mouseenter')
    wrapper.find('[data-test-id="day-current"]').at(0).simulate('mouseleave')
    wrapper.find('[data-test-id="day-current"]').at(0).simulate('click')
  })
})
