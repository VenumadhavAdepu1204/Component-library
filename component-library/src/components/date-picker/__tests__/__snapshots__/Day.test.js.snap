// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day  renders properly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "sc-gZMcBi eGeKTo",
      "data-test-id": "day-remaining",
    },
    "children": Array [
      Object {
        "data": "5",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "button",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "data-test-id": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "data-test-id": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
